recipes/automated-testing.md

```diff
@@ -0,0 +1,85 @@
+# Automated Testing
+
+## Types of Automated Tests
+
+There are different approaches to testing, and depending on the level of the
+entry point, we can split tests into following categories.
+
+- **Unit Tests**
+- **Integration Tests**
+- **E2E Tests**
+- **Load/Performance Tests**
+- **Visual Tests**
+
+*Note that some people can call these tests by different names, but for Studion
+internal purposes, this should be considered the naming convention.*
+
+### Unit Tests
+
+These are the most isolated tests that we can write. They should take a specific
```

Comment:
Would it make sense to structure test type descriptions in terms of:
- when to use
- when not to use
- best practices
- antipatterns

For example:
```
### Unit Tests
A unit can be a function, class, or module.
Unit testing is recommended for functions that contain a lot of logic and/or branching.
It is convenient to test a specific function at the lowest level so if the logic
changes, we can make minimal changes to the test suite and/or mocked data.

#### When to use: 
- Test a unit that implements the business logic, that's isolated from side effects such as database interaction or HTTP request processing
- Test function or class method with multiple input-output permutations
#### When not to use:
- To test unit that integrates different application layers, such as persistence layer (database) or HTTP layer (see "Integration Tests")
#### Best practices
- Unit tests should execute fast (<50ms)
- Use mocks and stubs through dependency injection (method or constructor injection)
#### Antipatterns
- Mocking infrastructure parts such as database I/O - instead, revert the control by using the `AppService`, `Command` or `Query` to integrate unit implementing business logic with the infrastructure layer of the application
- Monkey-patching dependencies used by the unit - instead, pass the dependencies through the constructor or method, so that you can pass the mocks or stubs in the test
```

Reply:
I'm writing content of the top of my mind just to give example.